tests/test_render.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+/** Unit tests for the Micropub Rendering class.
+ */
+
+class MicropubRenderTest extends WP_UnitTestCase {
+
+	function test_create_checkin_autogenerates_checkin_text_with_content() {
+		$input = array(
+			'type' => array( 'h-entry' ),
+			'properties' => array(
+				'content' => array( 'something' ),
+				'checkin' => array( array(
+					'properties' => array(
+						'name' => array( 'Place' ),
+						'url' => array( 'http://place' ),
+					),
+				) ),
+			),
+		);
+		$post_content = Micropub_Render::generate_post_content( 'something', $input );
+
+		$this->assertEquals( "<p>Checked into <a class=\"h-card p-location\" href=\"http://place\">Place</a>.</p>\n" .
+			"<div class=\"e-content\">\nsomething\n</div>",
+			$post_content );
+	}
+
+	function check_econtent() {
+		$content = '\n<h1>HTML content!</h1><p>coolio.</p>\n';
+		$input = array(
+			'properties' => array(
+				'content' => array( $content )
+			) );
+		$post_content = Micropub_Render::generate_post_content( 'something', $input );
+		$this->assertEquals( "<div class=\"e-content\">\n<h1>HTML content!</h1><p>coolio.</p>\n</div>", $post_content );
+	}
+
+	function create_interaction_json( $property ) {
```

Comment:
minor nit: remove `_json`, since this is after decoding the input data, ie not json specific.